### PR TITLE
fix(Core/Cryptography): OpenSSL 3.0 support

### PR DESCRIPTION
**Pull request by Kitzunu**
## Changes Proposed:
-  > It is the intention that the large majority of applications will work unchanged with OpenSSL 3.0 if those applications previously worked with OpenSSL 1.1.1. 

> Upgrading to OpenSSL 3.0 from OpenSSL 1.1.1 should be relatively straight forward in most cases. The most likely area where you will encounter problems is if you have used low level APIs in your code (as discussed above). In that case you are likely to start seeing deprecation warnings when compiling your application. If this happens you have 3 options:
>
>1) Ignore the warnings. They are just warnings. The deprecated functions are still present and you may still use them. However be aware that they may be removed from a future version of OpenSSL.
>
>2) Suppress the warnings. Refer to your compiler documentation on how to do this.
>
>3) Remove your usage of the low level APIs. In this case you will need to rewrite your code to use the high level APIs instead.

https://wiki.openssl.org/index.php/OpenSSL_3.0
- This somehow solves OpenSSL 3.0 builds.

## Issues Addressed:

- Closes 

## Tests Performed:

- Builds with 3.0

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Review by Nyeriah — APPROVED** [review]

File: src/server/game/Addons/AddonMgr.h [hunk 1]
```diff
@@ -23,6 +23,10 @@
 #include <string>
 #include <utility>
 
+#if AC_COMPILER == AC_COMPILER_GNU
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations" // In current OpenSSL 3.x MD5 is still a thing, so we use this to pass mac CI.
+#endif
```
Comment (Winfidonarleyan) [inline src/server/game/Addons/AddonMgr.h:28]: Avoid adding this to `.h' files

Comment (Winfidonarleyan) [reply src/server/game/Addons/AddonMgr.h:28]: Also, it is best to do this not for the whole file, but for some lines

**Comment by Winfidonarleyan** [comment]
https://github.com/TrinityCore/TrinityCore/commit/bc87f7b337154e683369a3790ee8fd1a7d4cba98

**Comment by mooreatv** [comment]
Curious why this was closed? it'd be nice to compile out of the box on Ubuntu 22 / openssl 3 no?

**Comment by rbedfordpro** [comment]
I also just got bit by this one when upgrading to Ubuntu 22. Now am unable to compile/run the server.

**Comment by Winfidonarleyan** [comment]
https://github.com/WarheadCore/WarheadBand/commit/d1172fd35f28e5a0731b6e8c0d577ba3f008324c

**Comment by MellianStudios** [comment]
@Winfidonarleyan im trying ur latest commit and server compiles and runs but when i attempt to log in game it crashes with
AC> #----------------------------------------------------------------------#
 
>> ASSERTION FAILED

# Location '/home/rymercyble/wow/azerothcore/src/common/Cryptography/ARC4.cpp:26'
# Function 'ARC4'
# Condition 'result == 1'
 
#----------------------------------------------------------------------#
Segmentation fault (core dumped)

**Pull request closed without merging**